runtime/autoload/health.vim

```diff
@@ -0,0 +1,413 @@
+  let pattern = '.*\%(\s\+\|\_$\)\zs\%<'.a:width.'c'
+  return map(split(a:text, pattern), 's:trim(v:val)')
+endfunction
+
+
+" Echo wrapped notes
+function! s:echo_notes(notes) abort
+  if empty(a:notes)
+    return
+  endif
+
+  echo '  Messages:'
+  for msg in a:notes
+    let msg_lines = s:textwrap(msg, 74)
+    echo '    *' msg_lines[0]
+    if len(msg_lines) > 1
+      echo join(map(msg_lines[1:], '"      ".v:val'), "\n")
+    endif
+  endfor
+endfunction
+
+
+" Load the remote plugin manifest file and check for unregistered plugins
+function! s:diagnose_manifest() abort
+  echo 'Checking: Remote Plugins'
+  let existing_rplugins = {}
+
+  for item in remote#host#PluginsForHost('python')
+    let existing_rplugins[item.path] = 'python'
+  endfor
+
+  for item in remote#host#PluginsForHost('python3')
+    let existing_rplugins[item.path] = 'python3'
+  endfor
+
+  let require_update = 0
+  let notes = []
+
+  for path in map(split(&rtp, ','), 'resolve(v:val)')
+    let python_glob = glob(path.'/rplugin/python*', 1, 1)
+    if empty(python_glob)
+      continue
+    endif
+
+    let python_dir = python_glob[0]
+    let python_version = fnamemodify(python_dir, ':t')
+
+    for script in glob(python_dir.'/*.py', 1, 1)
+          \ + glob(python_dir.'/*/__init__.py', 1, 1)
+      let contents = join(readfile(script))
+      if contents =~# '\<\%(from\|import\)\s\+neovim\>'
+        if script =~# '/__init__\.py$'
+          let script = fnamemodify(script, ':h')
+        endif
+
+        if !has_key(existing_rplugins, script)
+          let msg = printf('"%s" is not registered.', fnamemodify(path, ':t'))
+          if python_version == 'pythonx'
+            if !has('python2') && !has('python3')
+              let msg .= ' (python2 and python3 not available)'
+            endif
+          elseif !has(python_version)
+            let msg .= printf(' (%s not available)', python_version)
+          else
+            let require_update = 1
+          endif
+
+          call add(notes, msg)
+        endif
+
+        break
+      endif
+    endfor
+  endfor
+
+  echo '  Status: '
+  if require_update
+    echon 'Out of date'
+    call add(notes, 'Run :UpdateRemotePlugins')
+  else
+    echon 'Up to date'
+  endif
+
+  call s:echo_notes(notes)
+endfunction
+
+
+function! s:diagnose_python(version) abort
+  let python_bin_name = 'python'.(a:version == 2 ? '' : '3')
+  let pyenv = resolve(exepath('pyenv'))
+  let pyenv_root = exists('$PYENV_ROOT') ? resolve($PYENV_ROOT) : ''
+  let venv = exists('$VIRTUAL_ENV') ? resolve($VIRTUAL_ENV) : ''
+  let host_prog_var = python_bin_name.'_host_prog'
+  let host_skip_var = python_bin_name.'_host_skip_check'
+  let python_bin = ''
+  let python_multiple = []
+  let notes = []
+
+  if exists('g:'.host_prog_var)
+    call add(notes, printf('Using: g:%s = "%s"', host_prog_var, get(g:, host_prog_var)))
+  endif
+
+  let old_skip = get(g:, host_skip_var, 0)
+  let g:[host_skip_var] = 1
```

Comment:
If you set skip here, `Detect` will never return info like (when plugin is not installed):

```
provider/pythonx: Could not load Python 3:                                                                                                                     
/usr/bin/python3 does not have the neovim module installed. See ":help nvim-python".                                                                           
/usr/bin/python3.5 does not have the neovim module installed. See ":help nvim-python".                                                                         
python3.4 not found in search path or not executable.                                                                                                          
python3.3 not found in search path or not executable.                                                                                                          
/usr/bin/python does not have the neovim module installed. See ":help nvim-python".                                                                            
```

I believe this is quite useful to figure out which Pythons Neovim actually checked.